### ISSUE TITLE
Would be nice to add protection against redirect loops

### ISSUE THREAD

**Issue by dpovshed** [dpovshed/octopus#30]

**Comment by holtkamp** [comment]
@dpovshed can you elaborate on this? Something like:

- Site A redirects to Site B
- Site B redirects to Site A

??

We could keep track of "already processed" URLs...?

**Comment by dpovshed** [comment]
Yeah, even within a single site cyclic redirect can bring the app down!

What I think we can do - it is a) either track all visited URLs and check, or, b) just count a number of redirects and introduce a cap - like, not more than 10 redirects.

I like more approach (b) because: should be simpler; protect the cases with redirect glitches, when each new redirect is adding something to parameters - I had this case recently, like:
/user 
-> index.php?q=user 
-> index.php?q=user%q=user 
-> index.php?q=user%q=user%q=user 
etc

What do you think?
